File: credits.py
import json
import os
import xml.etree.ElementTree as ET


BASE_DIR = os.path.dirname(os.path.abspath(__file__))
WIKI_JSON_TITLE = "Credits.json"


def clean_text(value):
    return (value or "").strip()


def read_xml_text(path):
    data = open(path, "rb").read()
    for encoding in ("utf-8-sig", "utf-8", "gb18030"):
        try:
            return data.decode(encoding)
        except UnicodeDecodeError:
            pass
    return data.decode("utf-8", errors="replace")


def convert_data():
    xml_text = read_xml_text(os.path.join(BASE_DIR, "metas", "credits.xml"))
    root = ET.fromstring(xml_text)
    categories = []

    for category in root.findall("category"):
        name = clean_text(category.get("name"))
        entries = [clean_text(entry.text) for entry in category.findall("entry")]
        entries = [entry for entry in entries if entry]
        if not name and not entries:
            continue
        categories.append({
            "name": name,
            "entries": entries,
        })

    return categories


def convert():
    return json.dumps(convert_data(), ensure_ascii=False, separators=(",", ":"))

File: credits.lua
local p = {}

local dataCache

local function loadData()
    if dataCache then
        return dataCache
    end
    local ok, data = pcall(mw.loadJsonData, 'Credits.json')
    if not ok or type(data) ~= 'table' then
        local title = mw.title.new('Credits.json')
        if not title or not title.exists then
            return nil
        end
        ok, data = pcall(mw.text.jsonDecode, title:getContent() or '')
        if not ok or type(data) ~= 'table' then
            return nil
        end
    end
    dataCache = data.categories or data
    return dataCache
end

local function categoryText(category)
    if type(category) ~= 'table' then
        return nil
    end

    local name = category.name or ''
    local entries = category.entries or {}
    local result = {';' .. name}

    if type(entries) == 'table' then
        for _, entry in ipairs(entries) do
            if entry ~= nil and entry ~= '' then
                table.insert(result, ':' .. tostring(entry))
            end
        end
    end

    return table.concat(result, '\n')
end

function p.getCredits(frame)
    local data = loadData()
    if not data then
        return '错误：无法加载 [[Credits.json]]'
    end

    local result = {}
    for _, category in ipairs(data) do
        local text = categoryText(category)
        if text then
            table.insert(result, text)
        end
    end
    return table.concat(result, '\n')
end

function p.main(frame)
    return p.getCredits(frame)
end

return p